src/components/BottomWidgetsContainer.jsx

```diff
@@ -0,0 +1,63 @@
+import React       from 'react';
+import PropTypes   from 'prop-types';
+import { connect } from '../store/Connect';
+
+import '../../sass/components/_bottom-widget-container.scss';
+
+
+class BottomWidgetsContainer extends React.Component {
+    componentDidUpdate() {
+        if (React.Children.count(this.props.children)) {
+            this.props.updateChartMargin(200);
+        } else {
+            this.props.updateChartMargin(50);
+        }
+    }
+    render() {
+        const {
+            bottom,
+            isReadyToShow,
+            children,
+            top,
+        } =  this.props;
+
+        if (!isReadyToShow) {
+            return null;
+        }
+
+        const styles = {
+            top,
+            bottom,
+        };
+
+        return (
+            <div
+                className="cq-bottom-ui-widgets"
+                style={styles}
+            >
+                { children }
+            </div>
+        );
+    }
+}
+
+BottomWidgetsContainer.propTypes = {
+    bottom           : PropTypes.number,
+    isReadyToShow    : PropTypes.bool,
+    top              : PropTypes.number,
+    updateChartMargin: PropTypes.func.isRequired,
+};
+
+BottomWidgetsContainer.defaultProps = {
```

Comment:
@MahboobehMohammadi i think you should follow smart chart code style. its so confusing for next user to read different codes

Reply:
@MahboobehMohammadi regarding the discussion with Ashkan, we're planning to refactor the code style in this project in the next sprint so it's better to do them for the new component.